### PR TITLE
Add note to cluster-registration-url setting

### PR DESCRIPTION
**Pull request by w13915984028**
Related issue: https://github.com/harvester/harvester/issues/4944

### CODE REVIEW

File: docs/advanced/settings.md [hunk 1]
```diff
@@ -96,6 +96,22 @@ Default: none
 https://172.16.0.1/v3/import/w6tp7dgwjj549l88pr7xmxb4x6m54v5kcplvhbp9vv2wzqrrjhrc7c_c-m-zxbbbck9.yaml
 ```
 
+:::note
+
+When you set this item, a new POD from image `rancher/rancher-agent:related-version` will be created to do the registration work. The version of this image depends on the Rancher, it is not packed into Harvester ISO by default.
```
Comment (bk201) [inline docs/advanced/settings.md:101]: @w13915984028 Do you mean these cattle-cluster-agent pods in the host cluster? Here is a test if I disable my network and import Harvester to a Rancher instance:
截圖 2024-01-08 中午12 53 48

Perhaps what's needed here is to change the `containerd-registry` setting to point to a private registry.
Rancher's private registry setting seems to apply to guest clusters, not host guests. Can you help double check if the setting works, thanks!

Comment (w13915984028) [reply docs/advanced/settings.md:101]: Let me double check which/both registry setting will work.

Another question also comes to me: besides setting the registry, how can the user know the `related-version` of `rancher-agent` image ?  He needs to know the version first, then copy it the private registry.  Will check if Rancher has such documents to list it.

File: docs/advanced/settings.md [hunk 2]
```diff
@@ -96,6 +96,22 @@ Default: none
...
+
+Depends on the setting in your Rancher, the image will be downloaded from:
+
+(1) Rancher private registry.
+Please refer Rancher document to set the [global default private registry](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/global-default-private-registry).
+
+(2) Public docker image hub.
+This is the default registry when you do not set private rigistry in Rancher.
```
Comment (jillian-maroket) [inline docs/advanced/settings.md:109]:
```suggestion
- Docker Hub: This is the default option when you do not configure a private registry in Rancher.
```

File: docs/advanced/settings.md [hunk 3]
```diff
@@ -96,6 +96,22 @@ Default: none
...
+
+A last solution is to get the related image and upload it to all Harvester nodes.
```
Comment (jillian-maroket) [inline docs/advanced/settings.md:111]:
```suggestion
Alternatively, you can obtain a copy of the image and manually upload it to all Harvester nodes.
```

File: docs/advanced/settings.md [hunk 4]
```diff
@@ -96,6 +96,22 @@ Default: none
...
 ```
 
+:::note
+
+When you set this item, a new POD from image `rancher/rancher-agent:related-version` will be created to do the registration work. The version of this image depends on the Rancher, it is not packed into Harvester ISO by default.
+
+Depends on the setting in your Rancher, the image will be downloaded from:
+
+(1) Rancher private registry.
+Please refer Rancher document to set the [global default private registry](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/global-default-private-registry).
```
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply docs/advanced/settings.md:106]: We reserve ordered lists for procedures with multiple steps. When describing options, unordered lists are more appropriate.

**Review by jillian-maroket — COMMENTED** [review]
@w13915984028 Let me know if you have concerns about the changes.

File: docs/advanced/settings.md (hunk 1 above)
Comment (ibrokethecloud) [reply docs/advanced/settings.md:101]: the `rancher-images.txt` contains all the information for images used by rancher in an airgapped environment. The information and helper script is shipped with the release artifacts: https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/other-installation-methods/air-gapped-helm-cli-install/publish-images

Comment (w13915984028) [reply docs/advanced/settings.md]: Harvester `containerd-registry` is working; but `Rancher` setting seemed not affecting the image naming of `cattle-cluster-agent`.

**Review by ibrokethecloud — APPROVED** [review]
lgtm. thanks.

**Review by bk201 — APPROVED** [review]

**Review by jillian-maroket — COMMENTED** [review]
@w13915984028 I just cleaned up the parts that changed since the last review.

File: docs/advanced/settings.md [hunk 5]
```diff
@@ -96,6 +96,20 @@ Default: none
...
 
+:::note
+
+When you configure this setting, a new pod called `cattle-cluster-agent-*` in namespace `cattle-system` is created for registration purposes. This pod uses the container image `rancher/rancher-agent:related-version`. This image is not packed into the Harvester ISO and is instead determined by Rancher. In general, the `related-version` is same as Rancher version. For example, when register Harvester to a v2.7.9 Rancher, the image version is `rancher/rancher-agent:v2.7.9`. For further information, please refer to the [find the required assets for your rancher version](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/other-installation-methods/air-gapped-helm-cli-install/publish-images#1-find-the-required-assets-for-your-rancher-version)
+
+Depending on your Harvester settings, the image is downloaded from either of the following locations:
+
+ - Harvester containerd registry. Refer the next chapter `containerd-registry`.
+
+ - Docker Hub docker.io: This is the default option when you do not configure a private registry in Rancher.
```
[the remark this answers is not in the document]
Comment (w13915984028) [reply docs/advanced/settings.md]: Final as `Docker Hub (docker.io)`, the `docker.io` is the name used in pod container image like `"docker.io/rancher/rancher-agent:v2.7.6"`

**Review by jillian-maroket — APPROVED** [review]

**Pull request merged**